src/ProxyClient/MultiplexerClient.php

```diff
@@ -126,18 +160,51 @@ public function refresh($url, array $headers = [])
     }
 
     /**
-     * Helper function to invoke the given $method on all available ProxyClients implementing the given $interface.
+     * Invoke the given $method on all available ProxyClients implementing the
+     * given $interface.
      *
      * @param string $interface The FQN of the interface
      * @param string $method    The method to invoke
      * @param array  $arguments The arguments to be passed to the method
      */
     private function invoke($interface, $method, array $arguments)
     {
-        foreach ($this->proxyClients as $proxyClient) {
-            if (is_subclass_of($proxyClient, $interface)) {
-                call_user_func_array([$proxyClient, $method], $arguments);
-            }
+        foreach ($this->getProxyClients($interface) as $proxyClient) {
+            call_user_func_array([$proxyClient, $method], $arguments);
         }
     }
+
+    /**
+     * Invoke the given $method on the first available ProxyClient implementing
+     * the given $interface.
+     *
+     * @param string $interface The FQN of the interface
+     * @param string $method    The method to invoke
+     * @param array  $arguments The arguments to be passed to the method
+     *
+     * @return mixed Return value of ProxyClient method call
+     */
+    private function invokeFirst($interface, $method, array $arguments)
+    {
+        foreach ($this->getProxyClients($interface) as $proxyClient) {
+            return call_user_func_array([$proxyClient, $method], $arguments);
+        }
```

Comment:
what should we do when none of the clients implemented the requested interface? a bit hard to say - doing tags will not actually do anything, so we could throw an exception. on the other hand, we might just return a header name and an empty header value in that case, to not break users that check on the interface to know if they can use tags.

or we could create a TagCapableMultiplexerClient extends MultiplexerClient that will fail if none of its wrapped clients is TagCapable